### DISCUSSION TITLE
Move master node to other machine

### DISCUSSION THREAD

**Discussion by vinid223** [k3s-io/k3s#5311] [Q&A] [answered]
Hello there,

I have an odd use case. I do run a k3s cluster at home and I want to move the master node to a new machine.

I have a lot of volumes on longhorn and I really really don't want to move them manually. It is always a pain and a lot of setup. (As soon as I update Longhorn or do something else it just freaks out).

Is there a ways to just "backup" k3s data files including database and everything and restore it to a new machine ? Like not moving the OS or the binaries, but when installing k3s, it uses the existing data, keys, secrets, etc.

**Comment by brandond** [comment, marked as answer]
You can copy the stuff under `/var/lib/rancher/k3s/server`; that should bring over everything you need to use the same datastore, certs, et cetera. Pointing agents to the new `--server` URL should work fine; if not you can always switch back to the original.

**Reply by vinid223** [reply]
Great thanks, I'll try that

**Reply by Jmainguy** [reply]
This worked for me, thanks for posting.
